be/src/column/column_helper.h

```diff
@@ -280,19 +280,48 @@ class ColumnHelper {
      */
     template <LogicalType Type>
     static inline typename RunTimeColumnType<Type>::Ptr cast_to(const ColumnPtr& value) {
+#ifdef NDEBUG
         down_cast<const RunTimeColumnType<Type>*>(value.get());
+#else
+        auto* result = dynamic_cast<const RunTimeColumnType<Type>*>(value.get());
+        if (result == nullptr) {
+            const auto* ptr = value.get();
+            std::cerr << "Cast failed for column: "
+                      << " (expected type: " << Type << ", actual type: " << ptr->get_name() << ")" << std::endl;
+        }
+        assert(result != nullptr);
+#endif
         return RunTimeColumnType<Type>::static_pointer_cast(value);
     }
 
     template <LogicalType Type>
     static inline typename RunTimeColumnType<Type>::Ptr cast_to(ColumnPtr&& value) {
+#ifdef NDEBUG
         down_cast<const RunTimeColumnType<Type>*>(value.get());
+#else
+        auto* result = dynamic_cast<const RunTimeColumnType<Type>*>(value.get());
+        if (result == nullptr) {
+            const auto* ptr = value.get();
+            std::cerr << "Cast failed for column: "
+                      << " (expected type: " << Type << ", actual type: " << ptr->get_name() << ")" << std::endl;
+        }
+        assert(result != nullptr);
+#endif
         return RunTimeColumnType<Type>::static_pointer_cast(std::move(value));
     }
 
     template <LogicalType Type>
     static inline typename RunTimeColumnType<Type>::MutablePtr cast_to(MutableColumnPtr&& value) {
+#ifdef NDEBUG
         down_cast<const RunTimeColumnType<Type>*>(value.get());
+#else
+        auto* result = dynamic_cast<const RunTimeColumnType<Type>*>(value.get());
+        if (result == nullptr) {
+            std::cerr << "Cast failed for column: "
+                      << " (expected type: " << Type << ", actual type: " << value->get_name() << ")" << std::endl;
+        }
+        assert(result != nullptr);
```

Comment:
I think down_cast is enough

Reply:
We could consider modifying the implementation of down_cast directly when assert failed .

Reply:
yeah, that's a good idea, but not conflicted.
you can get the basic type info with `typeid`, but you can get more debug information if you already know it's a `Column`